### PR TITLE
Standardize artefact path for all build stages

### LINKED ISSUE #36414: Dockerfile: standardize artefact paths for intermediate stages
https://github.com/moby/moby/pull/35100 rewrote the Dockerfile to use multi-stage builds. Some review-comments were left.

A minor enhancement/cleanup: see https://github.com/moby/moby/pull/35100#discussion_r170234813

Can we standardise on an output path for artefacts? e.g. make all the build-stages output artefacts in (say) `/output/`, then we can simplify all these to

```Dockerfile
COPY --from=foo /output/ /usr/local/bin/
COPY --from=bar /output/ /usr/local/bin/
COPY --from=baz /output/ /usr/local/bin/
COPY --from=bif /output/ /usr/local/bin/
```

### PR DESCRIPTION
**Pull request by jeffhajewski**
Closes #36414 

**- What I did**
Updated the output path for all build artefacts to `/build/` in the Dockerfile

**- How I did it**
Changed the old output paths to `/build/`

**- How to verify it**
Easily verified by viewing the updated Dockerfile or the `/build/` directory of the intermediate builds

**- Description for the changelog**
Standarizes output path for artefacts of intermediate builds to `/build/`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -71,12 +71,12 @@ RUN set -x \
 	&& git clone https://github.com/docker/distribution.git "$GOPATH/src/github.com/docker/distribution" \
 	&& (cd "$GOPATH/src/github.com/docker/distribution" && git checkout -q "$REGISTRY_COMMIT") \
 	&& GOPATH="$GOPATH/src/github.com/docker/distribution/Godeps/_workspace:$GOPATH" \
-		go build -buildmode=pie -o /usr/local/bin/registry-v2 github.com/docker/distribution/cmd/registry \
+		go build -buildmode=pie -o /build github.com/docker/distribution/cmd/registry \
```
Comment (thaJeztah) [inline Dockerfile]: This should be `-o /build/registry-v2`

File: Dockerfile [hunk 2]
```diff
@@ -71,12 +71,12 @@ RUN set -x \
...
 	&& case $(dpkg --print-architecture) in \
 		amd64|ppc64*|s390x) \
 		(cd "$GOPATH/src/github.com/docker/distribution" && git checkout -q "$REGISTRY_COMMIT_SCHEMA1"); \
 		GOPATH="$GOPATH/src/github.com/docker/distribution/Godeps/_workspace:$GOPATH"; \
-			go build -buildmode=pie -o /usr/local/bin/registry-v2-schema1 github.com/docker/distribution/cmd/registry; \
+			go build -buildmode=pie -o /build github.com/docker/distribution/cmd/registry; \
```
Comment (thaJeztah) [inline Dockerfile]: This should be `-o /build/registry-v2-schema1`

File: Dockerfile [hunk 3]
```diff
@@ -99,15 +99,15 @@ RUN set -x \
 	&& export GOPATH="$(mktemp -d)" \
 	&& git clone https://github.com/go-swagger/go-swagger.git "$GOPATH/src/github.com/go-swagger/go-swagger" \
 	&& (cd "$GOPATH/src/github.com/go-swagger/go-swagger" && git checkout -q "$GO_SWAGGER_COMMIT") \
-	&& go build -o /usr/local/bin/swagger github.com/go-swagger/go-swagger/cmd/swagger \
+	&& go build -o /build github.com/go-swagger/go-swagger/cmd/swagger \
```
Comment (thaJeztah) [inline Dockerfile]: This should be `-o /build/swagger`

File: Dockerfile [hunk 4]
```diff
@@ -126,51 +126,51 @@ FROM base AS tomlv
 ENV INSTALL_BINARY_NAME=tomlv
 COPY hack/dockerfile/install/install.sh ./install.sh
 COPY hack/dockerfile/install/$INSTALL_BINARY_NAME.installer ./
-RUN PREFIX=/opt/$INSTALL_BINARY_NAME ./install.sh $INSTALL_BINARY_NAME
+RUN PREFIX=/build/$INSTALL_BINARY_NAME ./install.sh $INSTALL_BINARY_NAME
```
Comment (thaJeztah) [inline Dockerfile]: This should be `/build/` (without `$INSTALL_BINARY_NAME`) because it's the target-directory

File: Dockerfile [hunk 5]
```diff
@@ -126,51 +126,51 @@ FROM base AS tomlv
...
 
 FROM base AS vndr
 ENV INSTALL_BINARY_NAME=vndr
 COPY hack/dockerfile/install/install.sh ./install.sh
 COPY hack/dockerfile/install/$INSTALL_BINARY_NAME.installer ./
-RUN PREFIX=/opt/$INSTALL_BINARY_NAME ./install.sh $INSTALL_BINARY_NAME
+RUN PREFIX=/build/$INSTALL_BINARY_NAME ./install.sh $INSTALL_BINARY_NAME
```
Comment (thaJeztah) [inline Dockerfile]: Same here (and others below)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks! I left some comments 👍

**Comment by j-haj** [comment]
It appears two tests are failing:

* `DockerSwarmSuite.TestAPIServiceUpdatePort`
* `DockerSwarmSuite.TestSwarmPublishDuplicatePorts`

On the surface these don't seem like they should be related to the changes made to the Dockerfile. I'll poke around in `docker_cli_swar_test.go` to see if anything stands out.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**